kv_test.go

```diff
@@ -707,6 +715,84 @@ func TestSet5(t *testing.T) {
 	compareSlice(t, expected, values)
 }
 
+func TestSetIfPresent(t *testing.T) {
+	kv := NewKV[string](NewMapCache[string, string]())
+	kv.Set("a", "test2")
+	kv.Set("b", "test1")
+	kv.Set("c", "test4")
+
+	if !kv.SetIfPresent("a", "test5") {
+		t.Errorf("key \"abracadabra\" is present in kv, SetIfPresent should return true")
+	}
+
+	if !kv.SetIfPresent("a", "test6") {
+		t.Errorf("key \"abracadabra\" is present in kv, SetIfPresent should return true")
+	}
+
+	if kv.SetIfPresent("d", "test3") {
+		t.Errorf("key \"bbb\" is not present in kv, SetIfPresent should return false")
+	}
+
+	if kv.SetIfPresent("d", "test3") {
+		t.Errorf("key \"bbb\" is not present in kv, SetIfPresent should return false")
+	}
+
+	val, err := kv.Get("a")
+	if err != nil {
+		t.Fatalf("unexpected error in Get: %v", err)
+	}
+
+	if val != "test6" {
+		t.Errorf("expected %q, got %q", "test6", val)
+	}
+
+	_, err = kv.Get("d")
+	if err == nil {
+		t.Errorf("expected key \"d\" to not be present in the kv")
+	}
+}
+
+func TestSetIfPresentConcurrent(t *testing.T) {
+	kv := NewKV[string](NewMapCache[string, string]())
+	kv.Set("a", "startA")
+	kv.Set("b", "startB")
+
+	for i := 0; i < 1000; i++ {
+		go func() {
+			switch rand.Intn(6) {
+			case 0:
+				kv.SetIfPresent("a", "a")
+			case 1:
+				kv.SetIfPresent("b", "b")
+			case 2:
+				kv.SetIfPresent("c", "c")
+			case 3:
+				_, _ = kv.Get("a")
+				time.Sleep(5 * time.Millisecond)
```

Comment:
What does this sleep achieve?

Reply:
I'm not quite so sure anymore; it doesn't seem useful